django/views/generic/dates.py

```diff
@@ -687,7 +698,8 @@ def _get_next_prev(generic_view, date, is_previous, period):
         # Snag the first object from the queryset; if it doesn't exist that
         # means there's no next/previous link available.
         try:
-            result = getattr(qs[0], date_field)
+            result = functools.reduce(
```

Comment:
I think the `functool.reduce` suits nice in this case. Any other alternative will require more code, hence will add more complexity.